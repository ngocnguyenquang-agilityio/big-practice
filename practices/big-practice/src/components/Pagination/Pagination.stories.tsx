// Types
import type { Meta, StoryObj } from '@storybook/react';

// Components
import Pagination from '.';

const meta = {
  title: 'Components/Pagination',
  component: Pagination,
} satisfies Meta<typeof Pagination>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    totalPages: 4,
    standingPage: '1',
    handleChangePagination: () => {},
  },
};
