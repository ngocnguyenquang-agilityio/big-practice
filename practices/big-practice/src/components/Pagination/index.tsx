// Component
import { Button } from '@components/Button';

interface IPagination {
  totalPages: number;
  standingPage: null | string;
  handleChangePagination: (e: React.MouseEvent<HTMLButtonElement>) => void;
}

const Pagination = ({ totalPages, standingPage, handleChangePagination }: IPagination) => {
  return (
    <nav className='pt-4 flex justify-end'>
      <ul className='inline-flex -space-x-px text-md gap-4'>
        {Array.apply(0, new Array(totalPages)).map((_, idx) => {
          const currentPage = idx + 1;
          const isActivePage = currentPage.toString() === standingPage;

          return (
            <li key={idx}>
              <Button
                size='icon'
                variant='secondary'
                value={currentPage.toString()}
                onClick={handleChangePagination}
                className={isActivePage ? 'bg-blue-600' : ''}
              >
                {currentPage.toString()}
              </Button>
            </li>
          );
        })}
      </ul>
    </nav>
  );
};

export default Pagination;
