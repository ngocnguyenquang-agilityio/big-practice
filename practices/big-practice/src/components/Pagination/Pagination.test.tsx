import { RenderResult, fireEvent, render, screen } from '@testing-library/react';

// Components
import Pagination from './Pagination';

const props = {
  totalPages: 4,
  standingPage: '1',
  handleChangePagination: jest.fn(),
};

describe('Pagination Component', () => {
  let comp: RenderResult;

  beforeEach(() => {
    comp = render(<Pagination {...props} />);
  });

  test('Should render successfully', () => {
    expect(comp).toBeTruthy();
  });

  test('Should match snapshot', () => {
    expect(comp).toMatchSnapshot();
  });

  test('Should render correct number of pagination', async () => {
    const items = await screen.getByTestId('pagination');
    expect(items.children).toHaveLength(4);
  });

  test('Should get correct standing page', async () => {
    const items = await screen.getByTestId('pagination');
    expect(items.getElementsByClassName('bg-blue-600').length).toBe(1);
  });

  test('Should call handleChangePagination when user click', async () => {
    const button = screen.getByRole('button', { name: /2/i });
    fireEvent.click(button);
    expect(props.handleChangePagination).toBeCalledTimes(1);
  });
});
