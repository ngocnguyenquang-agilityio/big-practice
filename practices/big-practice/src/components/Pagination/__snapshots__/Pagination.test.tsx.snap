// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="pt-4 flex justify-end"
      >
        <ul
          class="inline-flex -space-x-px text-md gap-4"
          data-testid="pagination"
        >
          <li>
            <button
              class="inline-flex items-center text-white justify-center text-sm font-medium focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 border-neutral-800 ring-1 transition duration-300 ease-in-out enabled:hover:scale-110 enabled:hover:ring-blue-600 relative z-10 disabled:cursor-not-allowed overflow-hidden disabled:before:absolute disabled:before:inset-x-0 disabled:before:-z-10 disabled:before:h-px disabled:before:-rotate-45 disabled:before:transition-transform disabled:text-neutral-400 ring-neutral-700 disabled:before:bg-neutral-700 h-11 w-11 rounded-md bg-blue-600"
              value="1"
            >
              1
            </button>
          </li>
          <li>
            <button
              class="inline-flex items-center text-white justify-center text-sm font-medium focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 bg-neutral-900 border-neutral-800 ring-1 transition duration-300 ease-in-out enabled:hover:scale-110 enabled:hover:ring-blue-600 relative z-10 disabled:cursor-not-allowed overflow-hidden disabled:before:absolute disabled:before:inset-x-0 disabled:before:-z-10 disabled:before:h-px disabled:before:-rotate-45 disabled:before:transition-transform disabled:text-neutral-400 ring-neutral-700 disabled:before:bg-neutral-700 h-11 w-11 rounded-md"
              value="2"
            >
              2
            </button>
          </li>
          <li>
            <button
              class="inline-flex items-center text-white justify-center text-sm font-medium focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 bg-neutral-900 border-neutral-800 ring-1 transition duration-300 ease-in-out enabled:hover:scale-110 enabled:hover:ring-blue-600 relative z-10 disabled:cursor-not-allowed overflow-hidden disabled:before:absolute disabled:before:inset-x-0 disabled:before:-z-10 disabled:before:h-px disabled:before:-rotate-45 disabled:before:transition-transform disabled:text-neutral-400 ring-neutral-700 disabled:before:bg-neutral-700 h-11 w-11 rounded-md"
              value="3"
            >
              3
            </button>
          </li>
          <li>
            <button
              class="inline-flex items-center text-white justify-center text-sm font-medium focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 bg-neutral-900 border-neutral-800 ring-1 transition duration-300 ease-in-out enabled:hover:scale-110 enabled:hover:ring-blue-600 relative z-10 disabled:cursor-not-allowed overflow-hidden disabled:before:absolute disabled:before:inset-x-0 disabled:before:-z-10 disabled:before:h-px disabled:before:-rotate-45 disabled:before:transition-transform disabled:text-neutral-400 ring-neutral-700 disabled:before:bg-neutral-700 h-11 w-11 rounded-md"
              value="4"
            >
              4
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="pt-4 flex justify-end"
    >
      <ul
        class="inline-flex -space-x-px text-md gap-4"
        data-testid="pagination"
      >
        <li>
          <button
            class="inline-flex items-center text-white justify-center text-sm font-medium focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 border-neutral-800 ring-1 transition duration-300 ease-in-out enabled:hover:scale-110 enabled:hover:ring-blue-600 relative z-10 disabled:cursor-not-allowed overflow-hidden disabled:before:absolute disabled:before:inset-x-0 disabled:before:-z-10 disabled:before:h-px disabled:before:-rotate-45 disabled:before:transition-transform disabled:text-neutral-400 ring-neutral-700 disabled:before:bg-neutral-700 h-11 w-11 rounded-md bg-blue-600"
            value="1"
          >
            1
          </button>
        </li>
        <li>
          <button
            class="inline-flex items-center text-white justify-center text-sm font-medium focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 bg-neutral-900 border-neutral-800 ring-1 transition duration-300 ease-in-out enabled:hover:scale-110 enabled:hover:ring-blue-600 relative z-10 disabled:cursor-not-allowed overflow-hidden disabled:before:absolute disabled:before:inset-x-0 disabled:before:-z-10 disabled:before:h-px disabled:before:-rotate-45 disabled:before:transition-transform disabled:text-neutral-400 ring-neutral-700 disabled:before:bg-neutral-700 h-11 w-11 rounded-md"
            value="2"
          >
            2
          </button>
        </li>
        <li>
          <button
            class="inline-flex items-center text-white justify-center text-sm font-medium focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 bg-neutral-900 border-neutral-800 ring-1 transition duration-300 ease-in-out enabled:hover:scale-110 enabled:hover:ring-blue-600 relative z-10 disabled:cursor-not-allowed overflow-hidden disabled:before:absolute disabled:before:inset-x-0 disabled:before:-z-10 disabled:before:h-px disabled:before:-rotate-45 disabled:before:transition-transform disabled:text-neutral-400 ring-neutral-700 disabled:before:bg-neutral-700 h-11 w-11 rounded-md"
            value="3"
          >
            3
          </button>
        </li>
        <li>
          <button
            class="inline-flex items-center text-white justify-center text-sm font-medium focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 bg-neutral-900 border-neutral-800 ring-1 transition duration-300 ease-in-out enabled:hover:scale-110 enabled:hover:ring-blue-600 relative z-10 disabled:cursor-not-allowed overflow-hidden disabled:before:absolute disabled:before:inset-x-0 disabled:before:-z-10 disabled:before:h-px disabled:before:-rotate-45 disabled:before:transition-transform disabled:text-neutral-400 ring-neutral-700 disabled:before:bg-neutral-700 h-11 w-11 rounded-md"
            value="4"
          >
            4
          </button>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
